gw_eccentricity/eccDefinitionUsingAmplitude.py

```diff
@@ -53,6 +53,15 @@ def find_extrema(self, extrema_type="pericenters"):
         else:
             raise Exception("`extrema_type` must be either 'pericenters'"
                             " or 'apocenters'")
+        # We need to get the relevant segment from the full data only
+        # once, whereas `find_extrema` is called twice, once for the
+        # pericenters and then for the apocenters. We handle this
+        # by setting the flag `get_segment_of_data` to False after
+        # `self.get_segment_of_data_for_finding_extrema` is called for
+        # the first time.
+        if self.get_segment_of_data:
+            self.get_segment_of_data_for_finding_extrema()
+            self.get_segment_of_data = False
```

Comment:
Why not just do this immediately after the waveform is loaded? Does it need to be done in Amplitude and FrequencyFits

Reply:
We can not do this immediately after loading the waveform because we need to know the `tref_in` or `fref_in` for selecting segment around it. So, we can do this only inside `measure_ecc` which takes `tref_in` or `fref_in` as inputs. This also means that in the derived class when we initialize `self.data_for_finding_extrema = self.get_data_for_finding_extrema()` it uses the full waveform. Therefore, we need to handle the truncation of waveform separately before its used by `find_extrema`. One potential solution would be to move `tref_in`/`fref_in` to the arguments of the base class `eccDefiniton`. Should we do that?

Reply:
Moving to eccDefiniton might be ok. Is there a downside to doing it that way?

Reply:
Moved `tref_in/fref_in` to `self.__init__`. Now we get the segments in `self.__init__` only.